# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceParent.cpp
# Commit: 8d7f44b9b6e2
# Full Hash: 8d7f44b9b6e2debb08256460bc77355551e5d3c8
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-19 03:53:26
# Regressor Bug: 1839918
# File Overlap Count: 3
# Description:
#   Bug 1839918 - Create SetCookies variant for document.cookie on PWindowGlobal. r=bvandersloot,cookie-reviewers,valentin
#   
#   Using PWindowGlobal for SetCookies calls from content to parent process means we can get
#   the BrowsingContext the cookies are set for. The current implementation in PCookieService
#   drops this relationship.
# ==============================================================================

diff -r 7a28b63b4fc3 -r 8d7f44b9b6e2 netwerk/cookie/CookieServiceParent.cpp
--- a/netwerk/cookie/CookieServiceParent.cpp	Mon Sep 18 11:55:09 2023 +0000
+++ b/netwerk/cookie/CookieServiceParent.cpp	Mon Sep 18 11:55:09 2023 +0000
@@ -232,6 +232,14 @@
 IPCResult CookieServiceParent::RecvSetCookies(
     const nsCString& aBaseDomain, const OriginAttributes& aOriginAttributes,
     nsIURI* aHost, bool aFromHttp, const nsTArray<CookieStruct>& aCookies) {
+  return SetCookies(aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies,
+                    0, false);
+}
+
+IPCResult CookieServiceParent::SetCookies(
+    const nsCString& aBaseDomain, const OriginAttributes& aOriginAttributes,
+    nsIURI* aHost, bool aFromHttp, const nsTArray<CookieStruct>& aCookies,
+    uint64_t aBrowsingContextId, bool aIsThirdPartyCookie) {
   if (!mCookieService) {
     return IPC_OK();
   }
@@ -247,7 +255,8 @@
   mProcessingCookie = true;
 
   bool ok = mCookieService->SetCookiesFromIPC(
-      aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies, 0, false);
+      aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies,
+      aBrowsingContextId, aIsThirdPartyCookie);
   mProcessingCookie = false;
   return ok ? IPC_OK() : IPC_FAIL(this, "Invalid cookie received.");
 }